scalding-core/src/main/scala/com/twitter/scalding/typed/TypedPipe.scala

```diff
@@ -238,6 +238,9 @@ trait TypedPipe[+T] extends Serializable {
   /** adds a description to the pipe */
   def withDescription(description: String): TypedPipe[T] = new WithDescriptionTypedPipe[T](this, description)
 
+  /* add a config parameter to this pipe */
+  def withExtraConfig(key: String, value: String): TypedPipe[T] = new WithExtraConfig[T](this, key, value)
```

Comment:
What is the scope of this? There needs to be really precise contracts here to make a new API. I'm pretty nervous what this will mean, and that users will start coding to accidental implementation details which will be really hard to keep stable.